packages/core/src/types.ts

```diff
@@ -405,7 +426,10 @@ export interface StateNodeConfig<
    * The services to invoke upon entering this state node. These services will be stopped upon exiting this state node.
    */
   invoke?: SingleOrArray<
-    string | InvokeCreator<TContext, TEvent> | InvokeConfig<TContext, TEvent>
+    | string
+    | ActorCreator<TContext, TEvent>
```

Comment:
I must say I'm saddened by this - I think the design with "just actors" and dedicated "converters" was superior. It had nice decoupling, composition and tree-shaking characteristics. For instance while machine & promise actors might be considered really common, observables and callbacks are much less common - and right now every consumer is "paying" the cost of built support for those "integrations". 

Having all of those types decoupled from the core was a really good way of showing people that you actually can integrate easily **any** data/source type with XState, by simply writing a small adapter - in the very same way as "core" adapters were written.

Reply:
Hah, I actually made this change because you had concerns that there would be bigger breaking changes for the majority of users 😅 

Regardless, this was added back because `spawnFrom` is needed for inline `spawn()` (although this can also be done the same way, with the extra function call...)

Let's revisit this in a future PR

Reply:
> Hah, I actually made this change because you had concerns that there would be bigger breaking changes for the majority of users 😅

Well, I wasn't arguing against breaking changes per se - rather that we probably shouldn't advertise v5 as "no breaking changes" because that was misleading.

> Let's revisit this in a future PR

K, would be good to discuss this later on Slack to get a final decision on this, but I agree that it doesn't have to hold off this PR.

Keep in mind that we should be able to automate migration in many cases - it should be quite possible to provide an ESLint rule that would serve as a migration codemod.

Reply:
One happy medium would be:

```js
// ...
invoke: spawnFrom(promise /* or observable, callback, etc. */)
// ...
```

How does that sound?

Reply:
Yeah, I think this is a good idea - we should be able to provide fine-grained spawn creators (`spawnPromise`, `spawnObservable`, etc) but they could be wrapped for convenience within a `spawnFrom` helper.

All should be importable (and not injected) to aid tree-shaking.